transformer_engine/pytorch/module.py

```diff
@@ -269,11 +266,11 @@ def set_extra_state(self, state: torch.Tensor) -> None:
 
         # Restore global FP8 buffer states.
         set_global_fp8_buffer(state["global_fp8_buffer"])
-        set_global_fp8_recompute_buffer(state["global_fp8_recompute_buffer"])
-
         # Load extra items.
         self.fp8_meta.update(state["extra_fp8_variables"])
         self.fp8_meta["recipe"].amax_history_len = state["amax_history_fwd"].shape[0]
+        if "global_fp8_buffer_pos_fwd_recompute" in self.fp8_meta:
+            del self.fp8_meta["global_fp8_buffer_pos_fwd_recompute"]
```

Comment:
Now that we aren't storing the `_fp8_recompute_buffer` in the checkpoint, it is necessary that this key is not restored. If it is present, the tensor copy and restore functions in `fp8.py` for activation checkpointing will assume a non-empty buffer and index error will be thrown.